app/helpers/application_helper.rb

```diff
@@ -54,6 +54,15 @@ def render_index_grid(grid)
     render "admin/common/grid", no_records: no_records, grid: grid
   end
 
+  def render_loan_grid(grid)
+    if grid.all_pages_records.count == 1 && grid.filtered_by == ['id']
```

Comment:
As far as I understand how wice_grid works, is that it does the filtering on its own. `filter_by_params` is never called. Wice_grid wouldn't even let me access the 'all_pages_records' in the controller. `render_index_grid_with_redirect_check` is going to be the easier approach for sure. I could look into a filter based approach, but it is going to require some research first.